File: benchmarks/benchmark.dart
import 'dart:math';
import 'dart:io';

int fibonacci(int number) {
  if (number < 2) return number;
  return fibonacci(number - 1) + fibonacci(number - 2);
}

int countPrimes(int limit) {
  final isPrime = List<bool>.filled(limit + 1, true);
  isPrime[0] = false; isPrime[1] = false;
  for (var number = 2; number <= sqrt(limit); number++) {
    if (isPrime[number]) {
      for (var multiple = number * number; multiple <= limit; multiple += number) isPrime[multiple] = false;
    }
  }
  return isPrime.where((value) => value).length;
}

void main(List<String> arguments) {
  final fibonacciInput = int.tryParse(arguments[0]) ?? 35;
  final primeLimit = int.tryParse(arguments[1]) ?? 200000;
  var start = Stopwatch()..start(); final fibonacciResult = fibonacci(fibonacciInput); start.stop(); final fibonacciTime = start.elapsedMicroseconds / 1e6;
  start = Stopwatch()..start(); final primeResult = countPrimes(primeLimit); start.stop(); final primeTime = start.elapsedMicroseconds / 1e6;
  print('${fibonacciTime.toStringAsFixed(6)},${primeTime.toStringAsFixed(6)}');
}

File: benchmarks/benchmark.lua
function fibonacci(number)
  if number < 2 then return number end
  return fibonacci(number - 1) + fibonacci(number - 2)
end
function count_primes(limit)
  local is_prime = {}
  for number = 0, limit do is_prime[number] = true end
  is_prime[0] = false; is_prime[1] = false
  for number = 2, math.floor(math.sqrt(limit)) do
    if is_prime[number] then
      for multiple = number * number, limit, number do is_prime[multiple] = false end
    end
  end
  local count = 0
  for number = 0, limit do if is_prime[number] then count = count + 1 end end
  return count
end
local fibonacci_input = tonumber(arg[1]) or 35; local prime_limit = tonumber(arg[2]) or 200000
local start = os.clock(); local fibonacci_result = fibonacci(fibonacci_input); local fibonacci_time = os.clock() - start
start = os.clock(); local prime_result = count_primes(prime_limit); local prime_time = os.clock() - start
print(string.format("%.6f,%.6f", fibonacci_time, prime_time))
